platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/configuration/DaemonJvmToolchainSpec.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.launcher.daemon.configuration;
+
+import org.gradle.api.internal.provider.PropertyFactory;
+import org.gradle.api.provider.Property;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+import org.gradle.jvm.toolchain.JavaToolchainSpec;
+import org.gradle.jvm.toolchain.JvmImplementation;
+import org.gradle.jvm.toolchain.JvmVendorSpec;
+import org.gradle.jvm.toolchain.internal.DefaultJvmVendorSpec;
+
+import javax.annotation.Nonnull;
+
+public class DaemonJvmToolchainSpec implements JavaToolchainSpec {
```

Comment:
Definitely isn't necessary if we do agree on be able to specify `PropertyFactory` on the constructor by refactoring a bit the code. The only reason why was kept separately is because from `Launcher` context there's no access to `ObjectFactory`

Reply:
Can we just add another constructor to `DefaultToolchainSpec` that accepts a `PropertyFactory`? 

I would like to avoid two implementations here unless necessary